hs/src/Reach/Eval/Core.hs

```diff
@@ -2205,7 +2309,22 @@ doBalanceAssert_ fv mtok lhs op msg = do
       [cmp_v, public $ SLV_Bytes at msg]
 
 doBalanceUpdate :: Maybe DLArg -> PrimOp -> SLVal -> App ()
-doBalanceUpdate = doBalanceUpdate_ FV_balance
+doBalanceUpdate mtok op = \case
+  SLV_Int _ 0 -> return ()
+  rhs -> do
+    at <- withAt id
+    let up_rator = SLV_Prim $ SLPrim_PrimDelay at (SLPrim_op op) [] [(Public, rhs)]
+    -- Get the balance array
+    b <- lookupBalances
+    -- Find the current balance of mtok
+    bsv <- getBalanceOf mtok
+    -- Apply op to value
+    bv' <- evalApplyVals' up_rator [bsv]
+    bva <- compileCheckType T_UInt $ snd bv'
+    -- Set value/fluid ref with updated array
+    b' <- setBalance mtok bva
+    doFluidSet_ b b'
+
```

Comment:
This whole approach is interesting and different than I was thinking. I was thinking there would be `DLE_BalanceGet` and `DLE_BalanceSet` and `T_Balances` and then `FV_balances` of type `T_Balances` and then we'd transform them. But, thinking now, we might even get away with making `DLS_BalanceInit SrcLoc`, `DLS_BalanceGet SrcLoc DLVar DLArg`, and `DLS_BalanceSet SrcLoc DLVar DLArg DLArg` and then expand those into the array of the right size and uses of fluid set and ref. Then the rest of the compiler doesn't need to think about them at all and it happens before Linearize and we can discover the correct size right away.

Reply:
I will try this approach/what you described in slack